src/API/Configuration/Resolver/DotEnvResolver.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\API\Configuration\Resolver;
+
+/**
+ * Resolve variables from a `.env` file
+ * @psalm-internal \OpenTelemetry\API\Configuration
+ */
+class DotEnvResolver implements ResolverInterface
```

Comment:
The use-case I can think of is if you're using the SDK autoloading feature, including for auto-instrumentation. Since that all happens during composer autoloading, dotenv libraries won't have run yet.
That said, I don't have a strong opinion on having .env support or not, it just seemed like a quick win and potentially useful.

Reply:
The dotenv libraries seem to be more complex and handle more cases, I'm mainly concerned that the `DotEnvResolver` might only support a subset of the `.env` specification.